File: app-mvc-kotlin/src/main/java/com/naveed/samples/apputils/events/SingleDataEvent.java
package com.naveed.samples.apputils.events;

/**
 * Created by naveedali on 9/9/17.
 */

public class SingleDataEvent<T> implements BaseEvent {

    private SimpleEvent simpleEvent;
    public T data;


    public SingleDataEvent(boolean pStatus, int pId, String pMessage, T pData) {
        simpleEvent = new SimpleEvent(pStatus, pId, pMessage);
        this.data = pData;
    }


    @Override
    public int getEventId() {
        return simpleEvent.id;
    }

    @Override
    public boolean getStatus() {
        return simpleEvent.status;
    }

    @Override
    public String getMessage() {
        return simpleEvent.message;
    }
}

File: app/src/main/java/com/naveed/samples/apputils/events/SingleDataEvent.kt
package com.naveed.samples.apputils.events

/**
 * Created by naveedali on 9/9/17.
 */

class SingleDataEvent<T>(pStatus: Boolean, pId: Int, pMessage: String, var data: T) : BaseEvent {

    private val simpleEvent: SimpleEvent = SimpleEvent(pStatus, pId, pMessage)


    override val eventId: Int
        get() = simpleEvent.eventId

    override val status: Boolean
        get() = simpleEvent.status

    override val message: String
        get() = simpleEvent.message


}
